tests/test_call.py

```diff
@@ -13,10 +15,111 @@ class TestCall(unittest.TestCase):
+               "pagination":{
+                  "totalCount":0,
+                  "pageCount":0,
+                  "currentPage":0,
+                  "perPage":0
+               }
+            }
+            """
 
         call = Client('', http_client).call('call-id')
 
         http_client.request.assert_called_once_with('calls/call-id', 'GET', None)
 
-        self.assertEqual('ended', call.data.status)
\ No newline at end of file
+        self.assertEqual('ended', call.data.status)
+
+    def test_call_create(self):
+        api_response = {
+            "data": [
+                {
+                    "id": "21025ed1-cc1d-4554-ac05-043fa6c84e00",
+                    "status": "queued",
+                    "source": "31644556677",
+                    "destination": "31612345678",
+                    "createdAt": "2017-08-30T07:35:37Z",
+                    "updatedAt": "2017-08-30T07:35:37Z",
+                    "endedAt": None
+                }
+            ],
+            "_links": {
+                "self": "/calls/21025ed1-cc1d-4554-ac05-043fa6c84e00"
+            }
+        }
+
+        params = {
+            "source": "31644556677",
+            "destination": "31612345678",
+            "callFlow": {
+                "title": "Say message",
+                "steps": [
+                    {
+                        "action": "say",
+                        "options": {
+                            "payload": "This is a journey into sound. Good bye!",
+                            "voice": "male",
+                            "language": "en-US"
+                        }
+                    }
+                ]
+            },
+            "webhook": {
+                "url": "https://example.com",
+                "token": "token_to_sign_the_call_events_with",
+            }
+        }
+
+        http_client = Mock()
+        http_client.request.return_value = json.dumps(api_response)
+
+        call_creation_response = Client('', http_client).call_create(**params)
+
+        http_client.request.assert_called_once_with('calls', 'POST', params)
+
+        # check all api response data is outputted
+        expected_data = self.create_expected_call_data_based_on_api_response(api_response)
+        response_data = call_creation_response.data.__dict__
+        self.assertEqual(expected_data, response_data, 'Check client response contains the api response data.')
+
+        # check it can be formatted as string
+        expected_call_string = 'id                 : None\n' + \
+            'data.id                 : 21025ed1-cc1d-4554-ac05-043fa6c84e00\n' + \
+            'data.updatedAt          : 2017-08-30 07:35:37+00:00\n' + \
+            'data.createdAt          : 2017-08-30 07:35:37+00:00\n' + \
+            'data.endedAt            : None\n' + \
+            'data.webhook            : None'
+        self.assertEqual(expected_call_string, str(call_creation_response), 'Check returned call can be formatted as' +
```

Comment:
Maybe we can assert the individual properties? This way is very implementation specific to the `__str__` function?

Reply:
The individuals properties are checked above, this was just for checking I didn't break or forget anything in the `__str__` and also have it covered.